burr/core/parallelism.py

```diff
@@ -133,8 +135,59 @@ def run(
         )
         return state
 
+    async def _create_async_app(self, parent_context: ApplicationIdentifiers) -> Application:
+        builder = (
+            ApplicationBuilder()
+            .with_graph(self.graph.graph)
+            .with_spawning_parent(
+                app_id=parent_context.app_id,
+                sequence_id=parent_context.sequence_id,
+                partition_key=parent_context.partition_key,
+            )
+            # TODO -- handle persistence...
+            .with_identifiers(
+                app_id=self.application_id,
+                partition_key=parent_context.partition_key,  # cascade the partition key
+            )
+        )
+        if self.tracker is not None:
+            builder = builder.with_tracker(self.tracker)  # TODO -- move this into the adapter
+        # In this case we want to persist the state for the app
+        if self.state_persister is not None:
+            builder = builder.with_state_persister(self.state_persister)
+        # In this case we want to initialize from it
+        # We're going to use default settings (initialize from the latest
+        # TODO -- consider if there's a case in which we want to initialize
+        # in a custom manner
+        # if state_initializer is not None and self.cascade_state_initializer:
+        if self.state_initializer is not None:
+            builder = builder.initialize_from(
+                self.state_initializer,
+                default_state=self.state.get_all(),  # TODO _- ensure that any hidden variables aren't used...
+                default_entrypoint=self.graph.entrypoint,
+                resume_at_next_action=True,
+            )
+        else:
+            builder = builder.with_entrypoint(self.graph.entrypoint).with_state(self.state)
+
+        return await builder.abuild()
```

Comment:
OK, complexity, this has the off-chance of making something backwards incompatible. That said, I think that's OK TBH. In this case it's an odd one:
1. Someone is using the async parallelism piece
2. Someone is *not* using the async persister (which they won't be, it isn't there)
3. It breaks on reloading

I think very few people are doing this, but it *might* be worth having a parameter in `.abuild()` to bypass validation, we could do that here...

Reply:
We can add the parameter, but not sure if it is that helpful. If they are using sync persisters they will have to go through the sync `.build()` method anyway since `.abuild()` expects async persisters.

This is what I capture now  parallelism `.arun()`: we check if we have sync persisters and then use `.build()` to have that backward compatibility, otherwise everything async and ok.